### PR TITLE
Add Fibonacci sequence algorithm

### PR DESCRIPTION
**Pull request by MatanDavidi**
I added a Fibonacci sequence algorithm under "Algorithms/Numeric" and updated the README.md file to contain a reference to it

### CODE REVIEW

File: Algorithms/Numeric/Fibonacci.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            foreach (ulong element in Sequence)
+            {
+                Console.WriteLine(element);
+            }
+        }
+
+        /// <summary>
+        /// Generates the first n elements in the Fibonacci sequence, where n is equal to the value of <see cref="Size" />.
+        /// </summary>
+        private void GenerateFibonacci()
```
Comment (siriak) [inline Algorithms/Numeric/Fibonacci.cs]: This solution is very convoluted. Why don't just make a method called GetFibonacciSequence that returns IEnumerable and generates all Fibonacci numbers on demand? Or at least pass number of elements as an argument to this method and return them all (not print)

File: Algorithms/Numeric/Fibonacci.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        /// <summary>
+        /// Generates and returns the first n elements of the Fibonacci sequence.
+        /// </summary>
+        /// <param name="size">The number of elements in the sequence to generate.</param>
+        /// <returns>An IEnumerable of ulong variables containing the first elements of the Fibonacci sequence.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">This exception is thrown if a negative size is specified.</exception>
+        public static IEnumerable<ulong> GetSequence(int size)
+        {
+            List<ulong> sequence = new List<ulong>(size);
```
Comment (chuuddo) [inline Algorithms/Numeric/Fibonacci.cs]: I think you don't need to create a list here. 
Better to use `yield return` statement and then use `GetSequense().Take(size)`

Comment (siriak) [reply Algorithms/Numeric/Fibonacci.cs]: Right, generators are awesome! We won't even need to specify size here in that case.

File: Algorithms/Numeric/Fibonacci.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        /// <returns>An iterator that contains the Fibonacci sequence.</returns>
+        public static IEnumerable<ulong> GetSequence()
+        {
+            yield return 0;
+            yield return 1;
+            ulong a = 0;
+            ulong b = 1;
+            while (true)
+            {
+                yield return a + b;
```
Comment (siriak) [inline Algorithms/Numeric/Fibonacci.cs]:
```suggestion
                var temp = a + b;
                yield return temp;
```

File: Algorithms/Numeric/Fibonacci.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                ulong temp = a + b;
```
Comment (siriak) [inline Algorithms/Numeric/Fibonacci.cs]:
```suggestion
```

File: Algorithms/Numeric/Fibonacci.cs (hunk 3 above)
Comment (siriak) [reply Algorithms/Numeric/Fibonacci.cs]: So that we won't evaluate it twice.

File: Algorithms/Numeric/Fibonacci.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Algorithms.Numeric
+{
+    /// <summary>
+    /// The class Fibonacci handles the generation and printing of the fibonacci sequence up to its nth element.
+    /// For additional information, see https://en.wikipedia.org/wiki/Fibonacci_number.
+    /// </summary>
+    public abstract class Fibonacci
```
Comment (siriak) [inline Algorithms/Numeric/Fibonacci.cs]:
```suggestion
    public static class Fibonacci
```

**Pull request merged**